core/src/main/java/org/infinispan/security/impl/AuthorizationHelper.java

```diff
@@ -145,7 +145,7 @@ private SubjectACL computeSubjectACL(Subject subject, AuthorizationConfiguration
       int subjectMask = 0;
       Map<String, Role> globalRoles = globalConfiguration.authorization().roles();
       for (String role : allRoles) {
-         if (configuration == null || configuration.roles().contains(role)) {
+         if (configuration == null || configuration.roles().isEmpty() || configuration.roles().contains(role)) {
```

Comment:
I'm wondering if it would be better to have another method to be used when `configuration.roles()` is empty. 
Also, the subject mask can be computed in the same loop above. Example:

```
  for (Principal principal : principals) {
     Set<String> roleNames = roleMapper.principalToRoles(principal);
     if (roleNames != null) {
        for (String role : roleNames) {
           Role globalRole;
           if (allRoles.add(role) && (globalRole = globalRoles.get(role)) != null) {
              subjectMask |= globalRole.getMask();
           }
        }
     }
  }
```

Reply:
since the above is not related to this PR, I'll merge this PR.